### PR TITLE
Add support for IKEA Somrig shortcut button

### LINKED ISSUE #2835: New Ikea Somrig button, not able to connect zha
### Problem description

Hello :slight_smile:
I bought the new Ikea Somrig buttons and would like to connect them to my HomeAssistant with Zigbee Dongle.
During the search of a new device, HA identifies the button but is not able to configure it.
Although the configuration process does not finish, the device is part of the device list. Sometimes there is a battery and identify entity and sometimes not. I can not control any entity.

### Solution description

 Is anybody aware if there is a way to connect a device which is currently not fully supported or do you know if there will be an integration?

### Screenshots/Video

Bildschirm­foto 2023-12-12 um 20 17 57
Bildschirm­foto 2023-12-12 um 20 20 21

### Device signature

IEEE: 5c:c7:c1:ff:fe:87:35:e2
Nwk: 0x046e
Device Type: EndDevice
LQI: 108
RSSI: Unbekannt
Zuletzt gesehen: 2023-12-12T20:13:52
Energiequelle: Battery or Unknown

### Diagnostic information

[zha-e958911dd7b4b0c8793c91dc810d16d3-IKEA of Sweden SOMRIG shortcut button-7dfa14459bf34ecb1709bcf1e8686b31.json.txt](https://github.com/zigpy/zha-device-handlers/files/13652515/zha-e958911dd7b4b0c8793c91dc810d16d3-IKEA.of.Sweden.SOMRIG.shortcut.button-7dfa14459bf34ecb1709bcf1e8686b31.json.txt)

### Logs

<details><summary>Logs</summary>

```python
[Paste the logs here]
```

</details>

### Custom quirk

<details><summary>Custom quirk</summary>

```python
[Paste your custom quirk here]
```

</details>

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by casparz**
## Proposed change
Quirk for new SOMRIG smart button from IKEA. I based everything off of the Symfonysk remote v2.

## Additional information
Fixes #2835

## Checklist

- [x] The changes are tested and work correctly
- [x] `pre-commit` checks pass / the code has been formatted using Black
- [ ] Tests have been added to verify that the new code works

### CODE REVIEW

File: zhaquirks/ikea/somrigsmartbtn.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    device_automation_triggers = {
+        (PRESSED, BUTTON_1): {ENDPOINT_ID: 1, COMMAND: COMMAND_M_INITIAL_PRESS},
+        (SHORT_PRESS, BUTTON_1): {ENDPOINT_ID: 1, COMMAND: COMMAND_M_SHORT_RELEASE},
+        (DOUBLE_PRESS, BUTTON_1): {
+            ENDPOINT_ID: 1,
+            COMMAND: COMMAND_M_MULTI_PRESS_COMPLETE,
+        },
+        (LONG_PRESS, BUTTON_1): {ENDPOINT_ID: 1, COMMAND: COMMAND_M_LONG_PRESS},
+        (LONG_RELEASE, BUTTON_1): {ENDPOINT_ID: 1, COMMAND: COMMAND_M_LONG_RELEASE},
+
```
Comment (MattWestb) [inline zhaquirks/ikea/somrigsmartbtn.py]: No blank line in the code block so yo you is getting LINT error (delete line 157).

Comment (MattWestb) [reply zhaquirks/ikea/somrigsmartbtn.py]: Is the V2 commands working OK or have IKEA doing some changes in the cluster implantation (more presses then 2 is not implemented in the original and one command is having wrong data length) ?

Comment (casparz) [reply zhaquirks/ikea/somrigsmartbtn.py]: Thanks for reviewing, its my first contribution, and have to admit that I have just copied the Symfonisk2 quirk, and shaved away till it was working well. 

Can you elaborate on your question? I do not understand what you are asking.

**Comment by joshuapfl** [comment]
does it work now?

File: zhaquirks/ikea/somrigsmartbtn.py (hunk 1 above)
Comment (MattWestb) [reply zhaquirks/ikea/somrigsmartbtn.py]: If all is working OK then its great.
I can testing it later in the week then im back home.

Comment (joshuapfl) [reply zhaquirks/ikea/somrigsmartbtn.py]: Can you help me on how to ad?

created file in: zha_quirks/somrigsmartbtn.py

Copied your code in that file, and added path in config

zha:
   custom_quirks_path: /config/zha_quirks/
  
  Its not working. AM i missing something?  Does is require any other settings?

Comment (joshuapfl) [reply zhaquirks/ikea/somrigsmartbtn.py]: Using Home Assistant Supervised On Docker installtion
with instruction from here:
https://github.com/home-assistant/supervised-installer

Comment (joshuapfl) [reply zhaquirks/ikea/somrigsmartbtn.py]: ITS WORS
Didnt copy the code correctly :) need to go to RAW view....

Comment (joshuapfl) [reply zhaquirks/ikea/somrigsmartbtn.py]: THANK YOU

Comment (joshuapfl) [reply zhaquirks/ikea/somrigsmartbtn.py]: It works, but is shomehow slower other ikea remote, takes almost a second for it to trigger in HA. Any idea?

Comment (casparz) [reply zhaquirks/ikea/somrigsmartbtn.py]: I can confirm that it is slow, I thought this was normal for IKEA (the rest of my setup is mainly Sonoff and Hue).

Comment (joshuapfl) [reply zhaquirks/ikea/somrigsmartbtn.py]: Yes, the others are superfast, Expceilly for diming the delay is quite bad...

Comment (joshuapfl) [reply zhaquirks/ikea/somrigsmartbtn.py]: Checking the log, it first fires "Initial Press event was fired with parameters: {'new_position': 0}" and 1 sek later "Short Press - Button 1 event" tried to remove initial Press event in quirk and then it didnt work at all.
why does it need the initial event in the first place? 
Or is this just the nature of quirks being rather unresponsive?

Comment (MattWestb) [reply zhaquirks/ikea/somrigsmartbtn.py]: This device is using ZCL 9 (matter) commands that is not in Zigbbe ZCL but is coming in the next update of it.
Dont try deleting any commands then its braking  the logic that its being used in the future and its not bringing any good.
If like knowing how the logic is working download the later Matter ZCL and find the switch cluster and you is having all there.

Comment (joshuapfl) [reply zhaquirks/ikea/somrigsmartbtn.py]: Thanks for your answer. Ok. I will undelete it.
I am not understanding the rest you were sying 🤪 are you saying it’s possible to reduce that delay?
Appreciate your contribution

Comment (MattWestb) [reply zhaquirks/ikea/somrigsmartbtn.py]: No by the Matter ZCL its always sending one initial press and must waiting for more presses and then all is don its sending one last press so its not possible as long using this command sets (Aqara and other is not using standard commands so they is not certified Zigbee device).

Comment (joshuapfl) [reply zhaquirks/ikea/somrigsmartbtn.py]: I see :)
So somrig button is not really suitable for  accurate dimming. 
weird the other ikea buttons work just fine, very fast. I hope Ikea will not mess things up even further… maybe they don't want other people using it not in their way?
Unfortunately i couldn't buy older buttons. Somerig where the only ones in stock…

Comment (MattWestb) [reply zhaquirks/ikea/somrigsmartbtn.py]: **Way** using one Somrig  for dimmer then you can using Rodet and binding it to group / devices and its also working  then you host  system is off line **?**

Its like baying one large SUV and living in old town of Stockholm and you cant driving it and not parking it there =  better with one small Smart that working perfect for it.

Use the right device for the right function or you is only getting problems and Somrig is sending clicks for triggering things in automatons that is one online function and not autonomous Zigbee light commands that Roddret is doing (its one on/off dimmer controller).

Comment (joshuapfl) [reply zhaquirks/ikea/somrigsmartbtn.py]: Yes. I dont have any Rodrets unfortunately . They don't sell them here in Switzerland anymore. Only Somrig.

Comment (MattWestb) [reply zhaquirks/ikea/somrigsmartbtn.py]: "anymore"  = its only out of stocks then its the standard dimmer switch and shall being in all stores WW in January.
Its the same in Wien Wbh.

**Review by TheJulianJES — APPROVED** [review]
LGTM. Thanks!

**Comment by hugomn** [comment]
Hi @casparz! Is there any chance you could test the SOMRIG implementation again? As you can see [here](https://github.com/zigpy/zha-device-handlers/issues/2835#issuecomment-2404793590), it seems like no IKEA devices have been working anymore for a few weeks.

**Comment by MattWestb** [comment]
Deleted waiting one minute and adding it as new and i getting:
```
IKEA SOMRIG shortcut button Remote Button Double Press - Button 1 event was fired with parameters: {'previous_position': 0, 'total_number_of_presses_counted': 2}
15:44:31 - Now
```
Coordinator EZSP with blocked direct children.

**Pull request merged**